### PR TITLE
Improve advanced filter query support

### PR DESCRIPTION
**Pull request by niveathika**
## Purpose
Following diagnostics are added,
- [x]  Add error diagnostics when users use execute function
- [x] Add warning diagnostics when read function is used outside the scope of query clause

Following was improved,
- [x] Error diagnostics location was wrong after the query modification, this was fixed
- [x] Modifying of query syntax is now only done for persist clients unlike before which modifying if a remote call of  `read` is called

Related to: https://github.com/ballerina-platform/ballerina-standard-library/issues/3566

## Checklist
- [x] Linked to an issue
- [x] Updated the specification
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java [hunk 1]
```diff
@@ -69,15 +69,20 @@ public enum DiagnosticsCodes {
...
 
+    // todo: Add expression type in the diagnostic message
     PERSIST_201("PERSIST_201",
-            "Unsupported expression: read() function of persist client does not support this expression", ERROR),
+            "persist client read() function not supported for this expression", ERROR),
     PERSIST_202("PERSIST_202",
-            "Unsupported expression: read() function of persist client only supports int value in limit clause",
-            ERROR),
+            "persist client read() function limit not supported for variables, " +
+                    "limit expression should be an integer", ERROR),
```
Comment (daneshk) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java:77]:
```suggestion
            "persist client read() function limit expression doesn't support for variable references, " +
                    "it should be an integer value", ERROR),
```

File: compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java [hunk 2]
```diff
@@ -69,15 +69,20 @@ public enum DiagnosticsCodes {
     PERSIST_121("PERSIST_121", "unsupported features: {0} type is not supported",
             ERROR),
 
+    // todo: Add expression type in the diagnostic message
     PERSIST_201("PERSIST_201",
-            "Unsupported expression: read() function of persist client does not support this expression", ERROR),
+            "persist client read() function not supported for this expression", ERROR),
```
Comment (daneshk) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java:74]:
```suggestion
            "persist client read() function doesn't support for this query expression", ERROR),
```

File: compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java [hunk 3]
```diff
@@ -69,15 +69,20 @@ public enum DiagnosticsCodes {
...
+            "persist client read() function limit not supported for variables, " +
+                    "limit expression should be an integer", ERROR),
     PERSIST_203("PERSIST_203",
-            "Unsupported expression: read() function of persist client only supports selected variables "
-                    + "from the 'from clause' in order by clause",
-            ERROR);
+            "persist client read() function order by key should be ''variables'' in the ''from clause''",
+            ERROR),
+
+    PERSIST_210("PERSIST_210", "persist client execute() function is an internal function and cannot be invoked",
```
Comment (daneshk) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java:82]:
```suggestion
    PERSIST_210("PERSIST_210", "persist client execute() function is an internal function and external invocation is not allowed",
```

**Review by daneshk — DISMISSED** [review]
Add minor comments on the error message. Overall LGTM

File: compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java (hunk 1 above)
Comment (niveathika) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java]: Fixed in [4789703](https://github.com/ballerina-platform/module-ballerina-persist/pull/68/commits/4789703653ca02e339872b67b15944f2f061ab26)

File: compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java (hunk 2 above)
Comment (niveathika) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java]: Fixed in [4789703](https://github.com/ballerina-platform/module-ballerina-persist/pull/68/commits/4789703653ca02e339872b67b15944f2f061ab26)

File: compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java (hunk 3 above)
Comment (niveathika) [reply compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java]: Fixed in [4789703](https://github.com/ballerina-platform/module-ballerina-persist/pull/68/commits/4789703653ca02e339872b67b15944f2f061ab26)

**Review by daneshk — APPROVED** [review]

**Pull request merged**